web3/providers/ipc.py

```diff
@@ -136,6 +136,50 @@ def get_default_ipc_path(testnet=False):
         )
 
 
+def get_dev_ipc_path():
+    if sys.platform == 'darwin':
+        tmpdir = os.environ.get('TMPDIR', '')
+        ipc_path = os.path.expanduser(os.path.join(
+            tmpdir,
+            "geth.ipc"
+        ))
+        if os.path.exists(ipc_path):
+            return ipc_path
+
+    elif sys.platform.startswith('linux'):
+        ipc_path = os.path.expanduser(os.path.join(
+            "/tmp",
+            "geth.ipc"
+        ))
+        if os.path.exists(ipc_path):
+            return ipc_path
+
+    elif sys.platform == 'win32':
+        ipc_path = os.path.join(
+            "\\\\",
+            ".",
+            "pipe",
+            "geth.ipc"
+        )
```

Comment:
Geth opens the same IPC endpoint for dev and production on windows?

Reply:
On Windows Server 2016 I see the same IPC endpoint (geth.ipc) url populate in the console of ```geth --dev```. I have not seen the jsonrpc.ipc endpoint on dev or production windows, so I left it in there like ```get_default_ipc_path()``` had. I currently don't have access to another Windows variant to check endpoint paths.